eole/utils/loss.py

```diff
@@ -255,9 +255,11 @@ def ignore_prompt(self, batch):
             batch: The current batch.
         """
         # Create a mask with zeros at prompt positions and ones at answer postions.
-        mask = batch["src"].squeeze(dim=-1) == self.padding_idx
-        mask = torch.cumsum(mask.int(), 1)
-        # Apply the mask on the target side.
+        mask = batch["tgt"].squeeze(dim=-1) == self.padding_idx
+        mask = mask.cumsum(dim=1)
+        row_max = mask.max(dim=1, keepdim=True).values
+        mask = torch.where(mask < row_max, 0, mask)
+        mask = torch.where(mask >= row_max, 1, mask)
```

Comment:
I think you could use simpler stuff like:
mask = (batch["tgt"].squeeze(dim=-1) == self.padding_idx).cumsum(dim=1)
mask = (mask >= mask.max(dim=1, keepdim=True).values)
BUT CHECK FIRST

Reply:
It works fine !